resources/waterheater.rb

```diff
@@ -126,6 +126,17 @@ def self.apply_heatpump(model, runner, space, weather, t_set, vol, ef,
     pi = Math::PI
     r_tank = (UnitConversions.convert(v_actual, "gal", "m^3") / (pi * h_tank))**0.5
     a_tank = 2 * pi * r_tank * (r_tank + h_tank)
+
+    # water heater wrap calculation based on:
+    # Modeling Water Heat Wraps in BEopt DRAFT Technical Note
+    # Authors:  Ben Polly and Jay Burch (NREL)
+    if not jacket_r.nil?
+      a_side = 2 * pi * UnitConversions.convert(r_tank, "m", "ft") * UnitConversions.convert(h_tank, "m", "ft") # sqft
+      skin_insulation_t = 2.0 # inch
+      skin_insulation_R = 5.0 # R5
+      u_pre_skin = 1 / (skin_insulation_t * skin_insulation_R + 1.0 / 1.3 + 1.0 / 52.8) # Btu/hr-ft^2-F = (1 / hout + kins / tins + t / hin)^-1
```

Comment:
Should this be tied to the U value that we're already using? We already have a UA calculated earlier on in the HPWH method, I think we ought to be using that value here for "u_pre_skin"

Reply:
Is that only accounts for skin losses? Since the reason we cannot use UA calculated previously is that it not only accounts for skin but also shorts and flues. We kind of want to separate UA into UAskin, UAshorts, and UAflue and only apply jacket to the skin portion. (The ua in hpwh is hardcoded so I am not sure what it accounts for). But that ua value was used in the next line to be adjusted by subtracting something calculated as reflecting the jacket impact.

Reply:
OK, that's what I thought when I mentioned I might be misinterpreting things, the UA calculated previously is based on data from real HPWHs that were lab tested so it includes shorts (no flue in a HPWH). I think now that I understand this better I see the logic in what you're doing.